msrestazure/azure_active_directory.py

```diff
@@ -589,6 +590,43 @@ def get_msi_token(resource, port=50342):
     token_entry = result.json()
     return token_entry['token_type'], token_entry['access_token'], token_entry
 
+def get_msi_token_webapp(resource):
+    """Get a MSI token from inside a webapp or functions.
+
+    Env variable will look like:
+    MSI_ENDPOINT = http://127.0.0.1:41741/MSI/token/
+    MSI_SECRET = 69418689F1E342DD946CB82994CDA3CB
+    """
+    try:
+        msi_endpoint = os.environ['MSI_ENDPOINT']
```

Comment:
Small thing: this will never throw as the caller of `set_token` has checked the existence. You can custom the error message to target more towards missing MSI_ENDPOINT

Reply:
This can throw if you use the method directly, but yes it will never throw if you use `MSIAuthentication` class. But I want the method to be robust as well.